File: 039/happyNumbers.java
/*
	@author Sarah Kidd (www.linkedin.com/in/sarahkidd/)
	Description of problem at: https://www.codeeval.com/open_challenges/39/
*/

import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;

public class happyNumbers {
	public static void main(String[] args) {
		try {
			File file = new File(args[0]);
			BufferedReader infile = new BufferedReader(new FileReader(file));
			String line;
			while ((line = infile.readLine()) != null) {
				int num = Integer.parseInt(line);
				num = happyNums(num);
				System.out.println(num);
			}
			infile.close();
		} catch (FileNotFoundException e) {
			System.out.println("This is crap. Wrong stupid file name.");
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
	public static int happyNums(int num) {
		for (int i = 0; i < 100; i++) {
			int total = 0;
			while (num > 0) {
				total = (int)(Math.pow(num % 10,  2) + total);
				num = num / 10;
			}
			if (total == 1)
				return 1;
			num = total;
		}
		return 0;
	}
}

File: 039/happyNumbers.cs
/*
	@author Sarah Kidd (www.linkedin.com/in/sarahkidd/)
	Description of problem at: https://www.codeeval.com/open_challenges/39/
*/

using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sarahTesting
{
    class Program
    {
        static void Main(string[] args)
        {
            using (StreamReader reader = File.OpenText(args[0]))
                while (!reader.EndOfStream)
                {
                    string line = reader.ReadLine();
                    if (null == line)
                        continue;
                    int num = 0;
                    try
                    {
                        num = Convert.ToInt32(line);
                        num = happyNums(num);
                    }
                    catch (Exception exc)
                    {
                        Console.WriteLine("error occured: " + exc);
                    }
                    Console.WriteLine(num);
                }
        }
        public static int happyNums(int num)
        {
            for (int i = 0; i < 100; i++)
            {
                int total = 0;
                while (num > 0)
                {
                    total = Convert.ToInt32(Math.Pow((num % 10), 2)) + total;
                    num = Convert.ToInt32(num / 10);
                }
                if (total == 1)
                {
                    return 1;
                } 
                num = total;
            }
            return 0;
        }
    }
}
